podcasts/EpisodeDetailViewController.swift

```diff
@@ -552,6 +580,125 @@ private extension EpisodeDetailViewController {
+            .store(in: &cancellables)
+
+        addTabs()
+    }
+
+    private func addTabs() {
+        let tabContainerView = UIView()
+        tabContainerView.backgroundColor = .clear
+        tabContainerView.translatesAutoresizingMaskIntoConstraints = false
+
+        fakeNavView.addSubview(tabContainerView)
+
+        let trailingAnchor = tabContainerView.trailingAnchor.constraint(equalTo: fakeNavView.trailingAnchor)
+        NSLayoutConstraint.activate([
+            tabContainerView.leadingAnchor.constraint(equalTo: backBtn.trailingAnchor),
+            trailingAnchor,
+            tabContainerView.topAnchor.constraint(equalTo: backBtn.topAnchor),
+            tabContainerView.bottomAnchor.constraint(equalTo: backBtn.bottomAnchor)
+        ])
+
+        self.tabContainerView = tabContainerView
+        self.tabContainerTrailingAnchor = trailingAnchor
+
+        let viewModel = EpisodeTabsViewModel(tabs: [
+            .init(title: L10n.episodeDetailsTitle),
+            .init(title: L10n.bookmarks)
+        ])
+
+        let controller = ThemedHostingController(rootView: EpisodeDetailTabView(viewModel: viewModel))
+
+        tabContainerView.addSubview(controller.view)
+        controller.view.translatesAutoresizingMaskIntoConstraints = false
+        controller.view.anchorToAllSidesOf(view: tabContainerView)
+        addChild(controller)
+        controller.didMove(toParent: self)
+
+        tabViewModel = viewModel
+
+        // Listen for if the user taps the tab button directly
+        viewModel.$selectedTab
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] _ in
+                self?.selectedTabDidChange()
+            }
+            .store(in: &cancellables)
+    }
+
+    func selectedTabDidChange() {
+        guard let index = tabViewModel?.selectedIndex, let tab = Tab(rawValue: index), tab != currentTab else {
+            return
+        }
+
+        containerScrollView.scrollToPage(index)
+    }
+
+    private func didSwitchToTab(_ tab: Tab, animated: Bool = true) {
+        currentTab = tab
+        tabViewModel?.selectTabIndex(tab.rawValue)
+
+        guard animated else {
+            updateRightButtons()
+            return
+        }
+
+        UIView.animate(withDuration: 0.2) {
+            for button in self.rightActionButtons {
+                button.alpha = 0
+            }
+        } completion: { _ in
+            self.updateRightButtons()
+        }
+    }
+
+    private func updateRightButtons() {
+        removeAllButtons()
+
+        switch currentTab {
+        case .details:
+            addRightAction(image: UIImage(named: "podcast-share"), accessibilityLabel: L10n.share, action: #selector(shareTapped(_:)))
+            starButton = addRightAction(image: UIImage(named: "star_empty"), accessibilityLabel: L10n.starEpisode, action: #selector(starTapped(_:)))
+            updateStar()
+        case .bookmarks:
+            if bookmarksController?.viewModel.numberOfItems != 0 {
```

Comment:
Do we really need to handle this special case?

I would show the "..." anyways — we do this in Filters, for example. It doesn't matter if there are episode or not, it will be shown there.

I know the design doesn't show that but it looks like a lot of code for something minor. 🤷

Reply:
I don't think it's too bad, it's really just a couple of lines to listen for the change and reload the items.